internal/mempool/mempool.go

```diff
@@ -635,7 +620,7 @@ func (txmp *TxMempool) defaultTxCallback(req *abci.Request, res *abci.Response)
 		txmp.logger.Error(
 			"re-CheckTx transaction mismatch",
 			"got", wtx.tx.Hash(),
-			"expected", types.Tx(tx).Key(),
+			"expected", tx.Key(),
```

Comment:
Since this is now synchronous, this loop could be simplified. This loop operates on the understanding that an Async CheckTx could have failed. It keeps looping through the known Txs until it finds one matching the tx it cares about, skipping those that don't match. The non-matching are Tx that never received a response. Since the call is now synchronous, that condition should not happen, so I think this loop can be replaced with logic that just moves the 'cursor' in the case that a CheckTx errors.